package.json

```diff
@@ -21,7 +21,12 @@
     "serve": "docusaurus serve",
     "typecheck": "tsc",
     "write-translations": "docusaurus write-translations",
-    "write-heading-ids": "docusaurus write-heading-ids"
+    "write-heading-ids": "docusaurus write-heading-ids",
+    "build:validate-codeblocks": "tsc -p src/validate-codeblocks/tsconfig.json",
+    "validate-codeblocks": "pnpm run build:validate-codeblocks && node dist/validate-codeblocks/index.js",
```

Comment:
If we use tsx, you can run typescript files directly without having to precompile them